test/integration/features/experimental-flags.feature

```diff
@@ -0,0 +1,62 @@
+@experimental-flags
+Feature: Experimental Flags
+  Experimental flag --extra-clusterup-flags will be enabled by setting MINISHIFT_ENABLE_EXPERIMENTAL environment variable,
+  this flag will provide access to some of upcoming feature and experiments.
+
+  Scenario: User cannot start minishift experimental feature directly
+    Given Minishift has state "Does Not Exist"
+     Then executing "minishift start --service-catalog" fails
+      And stderr should contain
+      """
+      Error: unknown flag: --service-catalog
+      """
+
+  Scenario: User cannot use minishift flag --extra-clusterup-flag directly
+    Given Minishift has state "Does Not Exist"
+     Then executing "minishift start --extra-clusterup-flags" fails
+      And stderr should contain
+      """
+      Error: unknown flag: --extra-clusterup-flags
+      """
+
+  Scenario: User can enable and disable minishift experimental flag --extra-clusterup-flags
+    Given Minishift has state "Does Not Exist"
+     When setting up environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" with value "y" succeeds
+     Then executing "minishift start -h" succeeds
+      And stdout should contain
+      """
+      --extra-clusterup-flags string
+      """
+     When unset environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" succeeds
+     Then executing "minishift start -h" succeeds
+      And stdout should not contain
+      """
+      --extra-clusterup-flags string
+      """
+
+  Scenario: User cannot use minishift experimental flag --extra-clusterup-flag without experimental feature name
+    Given Minishift has state "Does Not Exist"
+     When setting up environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" with value "y" succeeds
+     Then executing "minishift start --extra-clusterup-flags" fails
+      And stderr should contain
+      """
+      Error: flag needs an argument: --extra-clusterup-flags
+      """
+      And unset environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" succeeds
+
+  Scenario: User can start minishift experimental feature
+    Given Minishift has state "Does Not Exist"
+      And image caching is disabled
+      And setting up environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" with value "y" succeeds
+     Then executing "minishift start --extra-clusterup-flags --service-catalog" succeeds
+      And stdout should contain
+      """
+      -- Extra 'oc' cluster up flags (experimental) ... 
+         '--service-catalog'
+      """
+      And unset environment variable "MINISHIFT_ENABLE_EXPERIMENTAL" succeeds
+
+  Scenario: Deleting Minishift
```

Comment:
Why do we need this scenario in this feature? or is it there in all the features.. to have a clean env for the next feature??

Reply:
@anjannath `Scenario: Deleting Minishift` is present in all features. It closes the minimal story of user, who starts Minishift, does something, deletes it in the end.

Reply:
This is a essential part of feature file. It completes the feature life cycle like ```start -> certain operation -> delete```. moreover the the delete scenario confirms that what ever the changes done in previous set of scenarios on same feature file should not break clean minishift delete.

> to have a clean env for the next feature??

Yes, it only deletes the vm, however the clean environment for the next feature run confirms by the function runCommandsBeforeFeature (https://github.com/minishift/minishift/blob/master/test/integration/testsuite/prepare.go#L154).

Reply:
Okay